### PR TITLE
Task B

### PR DESCRIPTION
**Pull request by hamza-sadruddin**
Please review the files for the necessary coding conventions. Thanks.

### CODE REVIEW

File: include/receiver_cadmium.hpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        // default constructor
+        Receiver() noexcept{
+            PREPARATION_TIME  = TIME("00:00:10");
+            state.ackNum      = 0;
+            state.sending     = false;
+        }
+        
+        // state definition
+        struct state_type{
+            int ackNum;
```
Comment (sdam12) [inline include/receiver_cadmium.hpp]: ackNum is a variable. It should all lower case according to Coding Convention 1.

File: include/receiver_cadmium.hpp [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            }    
+            return next_internal;
+        }
+
+        friend std::ostringstream& operator<<(std::ostringstream& os,
+                                              const typename Receiver<TIME>::state_type& i) {
+            os << "ackNum: " << i.ackNum; 
+        return os;
+        }
+    };
```
Comment (sdam12) [inline include/receiver_cadmium.hpp]: curly brackets should end in the same column.

File: include/subnet_cadmium.hpp [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+//This is a meta-model, it should be overloaded for declaring the "id" parameter
+template<typename TIME>
+class Subnet{
+    using defs = subnet_defs; // putting definitions in context
+    public:
+        //Parameters to be overwriten when instantiating the atomic model
+       
+        // default constructor
+        Subnet() noexcept{
+          state.transmiting     = false;
```
Comment (sdam12) [inline include/subnet_cadmium.hpp]: According to point 3. in coding convention Indentation: 4 spaces

File: include/receiver_cadmium.hpp [hunk 4]
```diff
@@ -0,0 +1,116 @@
+/**
+* Cristina Ruiz Martin
+* ARSLab - Carleton University
+*
+*/
+
+#ifndef _BOOST_SIMULATION_PDEVS_RECEIVER_HPP_
```
Comment (sdam12) [inline include/receiver_cadmium.hpp]: In each file according to rule 13. have header guards professor has given 2 guards at end and beginning. You have used only 1. I am not sure single is fine or we need double guards. Just noticed so mentioned in the review comments.
example: 
#ifndef __MY_HEADER__
#define __MY_HEADER__
... #endif // __ MY_HEADER__

File: include/data_structures/message.hpp [hunk 5]
```diff
@@ -10,22 +10,22 @@ using namespace std;
...
+    :value(i_value){}
 
-  	float   value;
+  	    float   value;
 
-  void clear() {
-   Message_t(); 
+    void clear() {
+        message_t(); 
   }
```
Comment (nick7695) [inline include/data_structures/message.hpp]: curly brackets should end in the same column.

File: src/main.cpp [hunk 6]
```diff
@@ -31,183 +31,224 @@ using TIME = NDTime;
...
+  struct oss_sink_provider{
+      static std::ostream& sink(){          
+          return out_data;
 
+        }
 
-/********************************************/
-/****** SUBNET *******************/
-/********************************************/
+    };
```
Comment (nick7695) [inline src/main.cpp:74]: In above lines check for indentation
Also curly brackets should end in the same column.

File: src/main.cpp [hunk 7]
```diff
@@ -31,183 +31,224 @@ using TIME = NDTime;
...
+
+    /************************/
+    /*******TOP MODEL********/
+    /************************/
+    cadmium::dynamic::modeling::Ports iports_TOP = {};
+    cadmium::dynamic::modeling::Ports oports_TOP = {typeid(outp_pack),typeid(outp_ack)};
+    cadmium::dynamic::modeling::Models submodels_TOP = {generator_con, ABPSimulator};
+    cadmium::dynamic::modeling::EICs eics_TOP = {};
+    cadmium::dynamic::modeling::EOCs eocs_TOP = {
+       cadmium::dynamic::translate::make_EOC<outp_pack,outp_pack>("ABPSimulator"),
```
Comment (nick7695) [inline src/main.cpp]: Check for indentation in this line and next line.

**Review by nick7695 — CHANGES_REQUESTED** [review]
Please check for comments and make necessary changes. Thank You.

**Review by sdam12 — CHANGES_REQUESTED** [review]
Review the comments and make required changes. Thank You.

File: include/receiver_cadmium.hpp (hunk 1 above)
Comment (hamza-sadruddin) [reply include/receiver_cadmium.hpp]: Fixed it. Thanks.

File: include/receiver_cadmium.hpp (hunk 2 above)
Comment (hamza-sadruddin) [reply include/receiver_cadmium.hpp]: Fixed it. Thanks.

File: include/subnet_cadmium.hpp (hunk 3 above)
Comment (hamza-sadruddin) [reply include/subnet_cadmium.hpp]: Fixed it. Thanks.

File: include/receiver_cadmium.hpp (hunk 4 above)
Comment (hamza-sadruddin) [reply include/receiver_cadmium.hpp]: I am also not sure regarding it. I looked it up online and found mixed opinions. Anyways, I added an additional underscore with all of them.

File: include/data_structures/message.hpp (hunk 5 above)
Comment (khanyousefzai) [reply include/data_structures/message.hpp]: Done as requested, Thanks

File: src/main.cpp (hunk 6 above)
Comment (khanyousefzai) [reply src/main.cpp:74]: Done as requested, Thanks

File: src/main.cpp (hunk 7 above)
Comment (khanyousefzai) [reply src/main.cpp]: Done as requested, Thanks

File: src/main.cpp [hunk 8]
```diff
@@ -31,183 +31,224 @@ using TIME = NDTime;
...
   auto start = hclock::now(); //to measure simulation execution time
 
 /*************** Loggers *******************/
-  static std::ofstream out_data("../data/abp_output.txt");
-    struct oss_sink_provider{
-        static std::ostream& sink(){          
-            return out_data;
-        }
+  static std::ofstream out_data("data/abp_output.txt");
+  struct oss_sink_provider{
```
Comment (sdam12) [inline src/main.cpp]: Look for indentation. I think you have provided only 2 spaces.

**Review by sdam12 — CHANGES_REQUESTED** [review]

**Review by sdam12 — COMMENTED** [review]
Changes requested looks ok.

**Review by sdam12 — COMMENTED** [review]
Changes in the last commit looks good. Thank You

**Review by sdam12 — APPROVED** [review]
Pull Request approved from my end.

**Review by nick7695 — APPROVED** [review]
Changes approved.

**Pull request merged**